proto/permissions.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+option go_package = "github.com/momentohq/client-sdk-go;client_sdk_go";
+option java_multiple_files = true;
+option java_package = "momento.shared.permissions";
+
+package permissions;
+
+// Aliases for categories of functionality.
+enum CacheRole {
+  CachePermitNone = 0;
+  // Restricts access to apis that read and write data from caches: No higher level resource description or modification.
+  CacheReadWrite = 1;
+  // Restricts access to apis that read from caches: No higher level resource description or modification.
+  CacheReadOnly = 2;
+  // Doesn't allow conditional write APIs (SetIfNotExists, IncreaseTTL etc)
+  CacheWriteOnly = 3;
+}
+
+// Aliases for categories of functionality.
+enum TopicRole {
+  TopicPermitNone = 0;
+  // Restricts access to apis that read and write data from topics: No higher level resource description or modification.
+  TopicReadWrite = 1;
+  // Restricts access to apis that read from topics: No higher level resource description or modification.
+  TopicReadOnly = 2;
+  // Only publish allowed
+  TopicWriteOnly = 3;
+}
+
+enum SuperUserPermissions {
+  SuperUser = 0;
+}
+
+message Permissions {
+  oneof kind {
+    SuperUserPermissions super_user = 1;
+    ExplicitPermissions explicit = 2;
+  }
+}
+
+message ExplicitPermissions {
+  repeated PermissionsType permissions = 1;
+}
+
+message PermissionsType {
+  oneof kind {
+    CachePermissions cache_permissions = 1;
+    TopicPermissions topic_permissions = 2;
+  }
+
+  message All {}
+
+  message CacheSelector {
+    oneof kind {
+      string cache_name = 1;
+    }
+  }
+
+  message CacheItemSelector {
+    oneof kind {
+      bytes key = 1;
+      bytes key_prefix = 2;
+    }
+  }
+
+  message CachePermissions {
```

Comment:
These message definitions are shared between regular v1 auth and acorn, however we don't actually want to support CacheItemSelector for v1 auth, right? Also, do we want to allow an acorn to be used for topics?
Perhaps it's clearer to have these restrictions in proto instead of mr2?

Reply:
For now, yes, the control-plane will reject generate v1 token request that include `CacheItemSelector`, but if we do want to support `CacheItemSelector` in v1 tokens later, i think it will be easier to make validation and sdk changes to allow rather then proto changes. Yes, we want to allow acorn's for us in topics, I think one of the key use cases for acorn's was browser chats.
i do think you're right that it's probably would be more clear, but I think the only groups that will have troubles are the brave souls using our proto's directly.